Trimble.Modus.Components/Controls/TMInput/CustomInput.xaml.cs

```diff
@@ -0,0 +1,551 @@
+                        SetBorderColor(customInput);
+                    }
+
+                }
+            }
+            else
+            {
+                // To avoid multiple times updating needs to check old value not equal to current value
+                if (customInput._validationResponse != ValidationResponse.Info)
+                {
+                    customInput._validationResponse = ValidationResponse.Info;
+                    SetBorderColor(customInput);
+                }
+            }
+            // customInput.TextChanged?.Invoke(customInput, new TextChangedEventArgs((string)oldValue, (string)newValue));
+        }
+    }
+
+    private static void SetBorderColor(CustomInput customInput)
+    {
+        if (!customInput.inputBorderlessEntry.IsFocused)
+        {
+            customInput.inputBorder.Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.Black);
+            customInput.inputBorder.StrokeThickness = 1;
+
+            if (!string.IsNullOrEmpty(customInput.HelperText))
+            {
+                customInput.inputHelperIcon.Source = ImageSource.FromFile(ImageConstants.HelperImage);
+                customInput.inputHelperLabel.Text = customInput.HelperText;
+            }
+        }
+        else
+        {
+            switch (customInput._validationResponse)
+            {
+                case ValidationResponse.Success:
+                    customInput.inputBorder.Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.Green);
+                    customInput.inputHelperIcon.Source = ImageSource.FromFile(ImageConstants.TickImage);
+                    customInput.inputBorder.StrokeThickness = 1;
+                    break;
+                case ValidationResponse.Error:
+                    customInput.inputBorder.Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.DangerRed);
+                    customInput.inputHelperIcon.Source = ImageSource.FromFile(ImageConstants.ErrorImage);
+                    customInput.inputBorder.StrokeThickness = 1;
+                    break;
+                default:
+                    customInput.inputBorder.Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+                    customInput.inputHelperIcon.Source = ImageSource.FromFile(ImageConstants.HelperImage);
+                    customInput.inputHelperLabel.Text = customInput.HelperText;
+                    customInput.inputBorder.StrokeThickness = 2;
+                    break;
+            }
+        }
+    }
+
+    private static void OnHelperTextChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable is CustomInput customInput)
+        {
+            SetBorderColor(customInput);
+        }
+    }
+
+    private void InputBorderlessEntry_Focused(object sender, FocusEventArgs e)
+    {
+        if (sender is BorderlessEntry)
+        {
+            SetBorderColor(_customInput);
```

Comment:
We can just pass `this` right? Do we have really have to hold `this` in a variable?

Reply:
Replaced with this